core/src/com/unciv/logic/map/mapgenerator/MapGenerator.kt

```diff
@@ -596,6 +651,146 @@ class MapGenerator(val ruleset: Ruleset) {
                     -1f, ruleset.modOptions.constants.spawnIceBelowTemperature,
                     0f, 1f))
             }.toList()
+
+        // Flat Earth needs a 1 tile wide perimeter of ice/mountain/snow and a 2 radius cluster of ice in the center.
+        if (tileMap.mapParameters.shape === MapShape.flatEarth) {
+
+            // Make sure center tiles and edge tiles are ocean not coast
+            for (tile in tileMap.values) {
+                val isCenterTile = tile.latitude == 0f && tile.longitude == 0f
+                val isEdgeTile = tile.neighbors.count() < 6
+
+                if (isCenterTile || isEdgeTile) {
```

Comment:
Please flip some if these ifs to reduce nesting...

Reply:
Like `if (!isCenterTile && !isEdgeTile) continue` ?

Reply:
Yup